lib/class-wp-rest-post-format-search-handler.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * REST API: WP_REST_Post_Format_Search_Handler class
+ *
+ * @package WordPress
+ * @subpackage REST_API
+ * @since 5.5.0
+ */
+
+/**
+ * Core class representing a search handler for post formats in the REST API.
+ *
+ * @since 5.5.0
+ *
+ * @see WP_REST_Search_Handler
+ */
+class WP_REST_Post_Format_Search_Handler extends WP_REST_Search_Handler {
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 5.0.0
+	 */
+	public function __construct() {
+		$this->type = 'post-format';
+	}
+
+	/**
+	 * Searches the object type content for a given search request.
+	 *
+	 * @since 5.0.0
+	 *
+	 * @param WP_REST_Request $request Full REST request.
+	 * @return array Associative array containing an `WP_REST_Search_Handler::RESULT_IDS` containing
+	 *               an array of found IDs and `WP_REST_Search_Handler::RESULT_TOTAL` containing the
+	 *               total count for the matching search results.
+	 */
+	public function search_items( WP_REST_Request $request ) {
+
+		$formats         = get_post_format_strings();
+		$format_fake_ids = array_keys( $formats );
+
+		if ( ! empty( $request['search'] ) ) {
+			$format_search = $request['search'];
+		}
+
+		$format_search = apply_filters( 'rest_post_format_search_query', $format_search, $request );
+
+		$found_ids = [];
+		foreach ( $format_fake_ids as $format_fake_id => $format_slug ) {
+			$format_title       = $formats[ $format_slug ];
+			$format_slug_match  = stripos( $format_slug, $format_search ) !== false;
+			$format_title_match = stripos( $format_title, $format_search ) !== false;
```

Comment:
Is there any way to include this in the actual query instead of matching in PHP?

Reply:
there is no query. formats are hardcoded strings